Dockerfile

```diff
@@ -64,149 +74,82 @@ ENV RENOVATE_BINARY_SOURCE=docker
-    rm Precompiled.zip
-
-ENV PATH $PATH:/opt/elixir-${ELIXIR_VERSION}/bin
+# renovate: datasource=docker versioning=docker
+RUN install-tool elixir 1.8.2
 
 # PHP Composer
 
-RUN echo "deb http://ppa.launchpad.net/ondrej/php/ubuntu bionic main" > /etc/apt/sources.list.d/ondrej-php.list && \
-    apt-key adv --keyserver keyserver.ubuntu.com --recv-keys 14AA40EC0831756756D7F66C4F4EA0AAE5267A6C && \
-    apt-get update && \
-    apt-get -y install php7.4-cli php7.4-mbstring php7.4-curl && \
-    rm -rf /var/lib/apt/lists/*
-
-# renovate: datasource=github-releases depName=composer/composer
-ENV COMPOSER_VERSION=1.10.5
-
-RUN php -r "copy('https://github.com/composer/composer/releases/download/$COMPOSER_VERSION/composer.phar', '/usr/local/bin/composer');"
-
-RUN chmod +x /usr/local/bin/composer
+# renovate: datasource=docker versioning=docker
+RUN install-tool php 7.4
 
-# Go Modules
+# renovate: datasource=github-releases lookupName=composer/composer
+RUN install-tool composer 1.10.5
 
-RUN apt-get update && apt-get install -y bzr mercurial && \
-    rm -rf /var/lib/apt/lists/*
+# Golang
 
-ENV GOLANG_VERSION=1.13.4
+# renovate: datasource=docker versioning=docker
+RUN install-tool golang 1.14.2
 
-# Disable GOPROXY and GOSUMDB until we offer a solid solution to configure
-# private repositories.
-ENV GOPROXY=direct GOSUMDB=off
-
-RUN wget -q -O go.tgz "https://golang.org/dl/go${GOLANG_VERSION}.linux-amd64.tar.gz" && \
-  tar -C /usr/local -xzf go.tgz && \
-  rm go.tgz && \
-  export PATH="/usr/local/go/bin:$PATH"
-
-ENV GOPATH /go
-ENV PATH $GOPATH/bin:/usr/local/go/bin:$PATH
+# Pip
 
-RUN mkdir -p "$GOPATH/src" "$GOPATH/bin" && chmod -R 777 "$GOPATH"
+# renovate: datasource=pypi
+RUN install-tool pip 20.0.2
 
-ENV CGO_ENABLED=0
+# Pipenv
 
-# Python
+# renovate: datasource=pypi
+RUN install-pip pipenv 2018.11.26
 
-RUN apt-get update && apt-get install -y python3.8-dev python3.8-venv python3-distutils && \
-    rm -rf /var/lib/apt/lists/*
+# Poetry
 
-RUN rm -fr /usr/bin/python3 && ln /usr/bin/python3.8 /usr/bin/python3
-RUN rm -rf /usr/bin/python && ln /usr/bin/python3.8 /usr/bin/python
+# renovate: datasource=github-releases lookupName=python-poetry/poetry
+RUN install-tool poetry 1.0.5
 
-# Pip
+# Cargo
 
-RUN curl --silent https://bootstrap.pypa.io/get-pip.py | python
+# renovate: datasource=docker versioning=docker
+RUN install-tool rust 1.36.0
 
 # CocoaPods
-RUN apt-get update && apt-get install -y ruby ruby2.5-dev && rm -rf /var/lib/apt/lists/*
-RUN ruby --version
-
-# renovate: datasource=rubygems depName=cocoapods versioning=ruby
-ENV COCOAPODS_VERSION 1.9.1
-RUN gem install --no-rdoc --no-ri cocoapods -v ${COCOAPODS_VERSION}
-
-
-# renovate: datasource=npm depName=npm versioning=npm
-ARG PNPM_VERSION=4.12.0
-RUN /usr/local/build/pnpm.sh
-
-USER ubuntu
 
-# HOME does not get passed after user switch :-(
-ENV HOME=/home/ubuntu
+# renovate: datasource=docker versioning=docker
+RUN install-tool ruby 2.5.8
```

Comment:
Is there a problem with running cocoapods with latest Ruby?

Reply:
not yet tested, want to make sure everything works as before, so renovate can propose the update